### DISCUSSION TITLE
Root Package Name

### DISCUSSION THREAD

**Discussion by LightBender** [dlang/PhobosV3-Design#27] [Polls]
While we are currently using `lib.*` it is still easy enough to change the root package name for Phobos and an intriguing option was suggested with the `d.*` idea. Please vote on which one you would prefer.

**Comment by dukc** [comment]
Voted `lib.` Reasoning: it actually describes the content. Library stuff, as opposed to the standard compiler API (`core`), implicit imports (`object`, admittedly badly named in this regard), or included foreign miscellania (`etc`).

`d.` on the other hand is descriptive only versus `etc`. Phobos is no more about D than `core` or `object` namespaces and therefore that name is a bit lacking IMO.

**Comment by atilaneves** [comment]
`phobos.`. "Lib" suggests it's just any other library, and it isn't. We have an advantage over other languanges in that the standard library actually has a name, and it's even a cool one! Let's use it.

**Comment by ichordev** [comment]
Q. Why `d.`?
A. Phobos is one of the most frequently used libraries, and has a few identical symbols between modules, so it should have a short name to help with writing out fully-qualified symbol names (i.e. `d.file.read` vs `d.stdio.read`). Phobos is also a major part of D's ecosystem. If you're not using Phobos then chances are it's because you're using BetterC, which—importantly—omits the name 'D'.

**Reply by jmdavis** [reply]
Just FYI, in a recent DLF planning meeting (shortly after Atila's comment here, I think), it was decided that the root name would be `phobos`. It is arguably a tad long (particularly when the module path gets deeper), but it's mostly only going to affect imports (as opposed to being in the middle of more complex expressions where the extra characters would matter more), and most everyone in that meeting (and Walter and Atila in particular) was very much in favor of `phobos`, so for better or worse, that's what it's going to be.
